airflow/contrib/operators/mlengine_operator.py

```diff
@@ -99,10 +40,26 @@ def _normalize_mlengine_job_id(job_id):
     Returns:
         A valid job_id representation.
     """
-    match = re.search(r'\d', job_id)
+
+    # Add a prefix when a job_id starts with a digit or a template
```

Comment:
We were calling this function at the Operator construction before, but now you call it within execute() (which I think is better).
Do you still get templates in the job_id? Shouldn't it be already replaced?

The most frequent errors we were getting around job_id while building these things up was due to setting job_id out of datetime. (which contains : or a space). 
If template is fed into this function, there's no use for this function as there's no guarantee on what characters that the template will be replaced to.

Reply:
You are right that the function doesn't make any sense if it allows a template to go thru all the way down to the job request.
I am not sure why this is happening honestly (not an expert on the template engine) because I was expecting it to be replaced but it wasn't happening in at object creation time and it didn't make any sense to me. So moving away that logic of the job to the execution of the operator and filtering there fixed it (and yes, there it gets properly filtered and no template gets passed down).